packages/toolpad-app/pages/app/[appId]/[version]/[[...path]].tsx

```diff
@@ -4,11 +4,12 @@ import { asArray } from '../../../../src/utils/collections';
 import ToolpadApp, { ToolpadAppProps } from '../../../../src/runtime/ToolpadApp';
 
 export const getServerSideProps: GetServerSideProps<ToolpadAppProps> = async (context) => {
-  const { loadRenderTree, parseVersion } = await import('../../../../src/server/data');
+  const { loadRenderTree, parseVersion, getApp } = await import('../../../../src/server/data');
 
   const [appId] = asArray(context.query.appId);
   const version = parseVersion(context.query.version);
-  if (!appId || !version) {
+  const app = appId ? await getApp(appId) : null;
+  if (!appId || !app || !version) {
```

Comment:
No need to check for `appId` here anymore?

Reply:
it's being used down below, this asserts it's defined

Reply:
ah nevermind